api/envoy/type/tracing.proto

```diff
@@ -0,0 +1,76 @@
+syntax = "proto3";
+
+package envoy.type;
+
+option java_outer_classname = "TracingProto";
+option java_multiple_files = true;
+option java_package = "io.envoyproxy.envoy.type";
+
+import "validate/validate.proto";
+
+// [#protodoc-title: Tracing]
+
+// Describes custom tags for the active span.
+message TracingCustomTag {
+  // Literal type custom tag with literal value for tag value.
+  message Literal {
+    // literal value to populate the tag value.
+    string value = 1 [(validate.rules).string = {min_bytes: 1}];
+  }
+
+  // Environment type custom tag with environment name and default value.
+  message Environment {
+    // environment variable name to obtain the value to populate the tag value.
+    string name = 1 [(validate.rules).string = {min_bytes: 1}];
+
+    // used to populate the tag value if the environment variable is not found.
+    string default_value = 2;
+  }
+
+  // Header type custom tag with header name and default value.
+  message Header {
+    // request header name to obtain the value to populate the tag value.
+    string name = 1 [(validate.rules).string = {min_bytes: 1}];
+
+    // used to populate the tag value if the name is not found in request header map.
+    string default_value = 2;
+  }
+
+  // Metadata type custom tag with namespace, path, and default value.
```

Comment:
Can you flesh out the docs for this including potentially a full YAML example? It's confusing to me with these docs to understand how this actually works, what path is, path_separator, etc.

Reply:
documented.